src/NCRings.jl

```diff
@@ -155,6 +155,27 @@ julia> is_unit(ZZ(-1)), is_unit(ZZ(4))
 """
 function is_unit end
 
+@doc raw"""
+    is_nilpotent(a::T) where {T <: NCRingElement}
+
+Return true iff $a$ is nilpotent, i.e. a^k == 0 for some k.
+
+# Examples
+```jldoctest
+julia> ZZ720,_ = residue_ring(ZZ,720);
+
+julia> S, x = polynomial_ring(QQ, :x);
+
+julia> is_nilpotent(30*x), is_nilpotent(30+90*x), is_nilpotent(S(15))
+(true, true, false)
```

Comment:
@JohnAAbbott seems you and the code disagree as to what the correct output is?

Reply:
Thepolynomial_ring should be over the ring created before, not over QQ. Fixed now